### DISCUSSION TITLE
Generic error structure

### DISCUSSION THREAD

**Discussion by mantzas** [aws/aws-sdk-go-v2#1868] [Q&A] [answered]
Hi,

I am migrating to v2 and trying to replace this piece of code:

```
if awsErr, ok := err.(awserr.Error); ok {
	// S3 will return an errCodeNotModified if the object was not modified (= same ETag).
	// Please note however that this will not work with Localstack: https://github.com/localstack/localstack/issues/1071
	if awsErr.Code() == errCodeNotModified {
		continue
	}

	if awsErr.Code() == errNoSuchKey && skipLogging[i] {
		continue
	}
}
```
I cannot find the equivalent of `awserr.Error`. Can you give me a hint?
Thanks

**Comment by RanVaknin** [comment, marked as answer]
Hi @mantzas ,

Please refer to the [Error handling section in the v2 migration guide](https://aws.github.io/aws-sdk-go-v2/docs/migrating/#errors-types).
It has that exact example of how to transform what you have there into v2 code.

Let me know if that helps.

Thanks,
Ran~
